test/operance_data_controller_test.dart

```diff
@@ -0,0 +1,444 @@
+      setUp(() async {
+        await initializeController(
+          initialPage: initialPageWithRows,
+        );
+      });
+
+      test('Then it should toggle the expansion of a row', () {
+        controller.toggleExpandedRow(0);
+        expect(controller.expandedRows, <int, bool>{0: true});
+
+        controller.toggleExpandedRow(0);
+        expect(controller.expandedRows, <int, bool>{0: false});
+      });
+    });
+
+    group('When hovering rows', () {
+      setUp(() async {
+        await initializeController(
+          initialPage: initialPageWithRows,
+        );
+      });
+
+      test('Then it should set the hovered row index', () {
+        controller.setHoveredRowIndex(1);
+        expect(controller.hoveredRowIndex, 1);
+      });
+
+      test('Then it should clear the hovered row index', () {
+        controller
+          ..setHoveredRowIndex(1)
+          ..clearHoveredRowIndex();
+        expect(controller.hoveredRowIndex, isNull);
+      });
+    });
+
+    group('When updating rows', () {
+      setUp(() async {
+        await initializeController(
+          initialPage: initialPageWithRows,
+        );
+      });
+
+      test('Then it should update the row in cache', () {
+        controller.updateRow('Row1');
+        expect(
+          controller.currentRows,
+          <String>['Row1', 'Row2'],
+        );
+      });
+    });
+
+    group('When getting visible columns', () {
+      test('Then it should return the visible columns', () async {
+        await initializeController(
+          columnOrder: columnOrder,
+          initialPage: initialPageWithRows,
+        );
+
+        controller.hideColumn(1);
+        expect(controller.visibleColumns, <int>{0, 2});
+      });
+    });
+
+    group('When hiding and showing columns', () {
+      test('Then it should hide and show columns correctly', () async {
+        await initializeController(
+          columnOrder: columnOrder,
+          initialPage: initialPageWithRows,
+        );
+
+        controller.hideColumn(1);
+        expect(controller.hiddenColumns, <int>{1});
+
+        controller.showColumn(1);
+        expect(controller.hiddenColumns, isEmpty);
+      });
+
+      test('Then it should toggle column visibility', () async {
+        await initializeController(
+          columnOrder: columnOrder,
+          initialPage: initialPageWithRows,
+        );
+
+        controller.toggleColumnVisibility(1);
+        expect(controller.hiddenColumns, <int>{1});
+
+        controller.toggleColumnVisibility(1);
+        expect(controller.hiddenColumns, isEmpty);
+      });
+
+      test('Then it should show all columns', () async {
+        await initializeController(
+          columnOrder: columnOrder,
+          initialPage: initialPageWithRows,
+        );
+
+        controller
+          ..hideColumn(1)
+          ..showAllColumns();
+        expect(controller.hiddenColumns, isEmpty);
+      });
+    });
+
+    group('When managing searched rows', () {
```

Comment:
These search tests are rather simplistic. Maybe include some of the following:

Testing search with different case sensitivity
Testing search with empty strings
Testing search with special characters
Testing search performance with large datasets